pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -740,3 +744,21 @@ func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
 	snapstoreHash[ssr.snapstoreConfig.Provider] = newSnapstoreSecretHash
 	return true
 }
+
+// IsScheduledFullSnapshotMissed checked whether the last scheduled full-snapshot is missed or scheduled full-snapshot can be missed.
+func (ssr *Snapshotter) IsScheduledFullSnapshotMissed() bool {
+	if time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() > recentFullSnapshotPeriodInHours {
+		return true
+	}
+
+	now := time.Now()
+	nextSnapSchedule := ssr.schedule.Next(now)
+	timeLeftToTakeNextSnap := nextSnapSchedule.Sub(now)
+
+	if miscellaneous.GetPrevDayScheduledSnapTime(nextSnapSchedule) == ssr.PrevFullSnapshot.CreatedOn {
+		return false
+	}
+
+	// to check whether schedule full-snapshot can be missed.
+	return timeLeftToTakeNextSnap.Hours()+time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() > recentFullSnapshotPeriodInHours
```

Comment:
Lets say the next snapshot time is in 5 mins. Will you still trigger a full snapshot? Once you trigger a full snapshot and then when the regular scheduled full snapshot time is reached then will yet another full snapshot be triggered?

Reply:
It's ok to have a 2 full snapshot rather then no full snapshot for >24hrs and having the alerts as well